src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,127 @@
+import { FC } from 'react';
+import { Link, useLocation, useParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { Person } from '../../types/Person';
+import { PersonLink } from '../PersonLink';
+import { getSearchWith } from '../../utils/searchHelper';
+
+const titleList = [
+  { name: 'Name', link: 'name' },
+  { name: 'Sex', link: 'sex' },
+  { name: 'Born', link: 'born' },
+  { name: 'Died', link: 'died' },
+];
+
+export interface Props {
+  people: Person[],
+  sort: string,
+  order: string,
+  searchParams: URLSearchParams,
+}
+
+export const PeopleTable: FC<Props> = ({
+  people,
+  sort,
+  order,
+  searchParams,
+}) => {
+  const { slug: selectedSlug } = useParams();
+  const location = useLocation();
+
+  const onSort = (value: string) => {
+    if (sort === value && !order) {
+      return {
+        search: getSearchWith((searchParams), { sort: value, order: 'desk' }),
+      };
+    }
+
+    if (sort === value && order) {
+      return {
+        search: getSearchWith((searchParams), { sort: null, order: null }),
+      };
+    }
+
+    return {
+      search: getSearchWith((searchParams), { sort: value, order: null }),
+    };
```

Comment:
Optimize this code. Do nested if to avoid code duplication. Try to use the ternary operator.